src/item/CustomiesItemFactory.php

```diff
@@ -61,28 +66,31 @@ public function getItemTableEntries(): array {
 	/**
 	 * Registers the item to the item factory and assigns it an ID. It also updates the required mappings and stores the
 	 * item components if present.
+	 * CreativeInventoryInfo not completely implemented yet.
 	 * @phpstan-param class-string $className
 	 */
 	public function registerItem(string $className, string $identifier, string $name): void {
-		if($className !== Item::class) {
+		if ($className !== Item::class) {
 			Utils::testValidInstance($className, Item::class);
 		}
 		/** @var Item $item */
 		$item = new $className(new ItemIdentifier(++$this->nextItemID, 0), $name);
 
-		if(ItemFactory::getInstance()->isRegistered($item->getId())) {
+		if (ItemFactory::getInstance()->isRegistered($item->getId())) {
 			throw new RuntimeException("Item with ID " . $item->getId() . " is already registered");
 		}
 		$this->registerCustomItemMapping($item->getId());
 		ItemFactory::getInstance()->register($item);
 
-		if(($componentBased = $item instanceof ItemComponents)) {
+		if (($componentBased = $item instanceof ItemComponents)) {
 			$componentsTag = $item->getComponents();
 			$componentsTag->setInt("id", $item->getId());
 			$componentsTag->setString("name", $identifier);
 			$this->itemComponentEntries[$identifier] = new ItemComponentPacketEntry($identifier, new CacheableNbt($componentsTag));
 		}
+
 		$this->itemTableEntries[$identifier] = new ItemTypeEntry($identifier, $item->getId(), $componentBased);
+		CreativeInventory::getInstance()->add($item);
```

Comment:
I'm not entirely sure if registering things in the creative inventory by default is a good idea. Is this customies' responsibility? @TwistedAsylumMC please check

Reply:
In fact, if you don't put a CreativeInventoryInfo it won't appear in any of the creative inventory boxes 

I already tested everything and sent screenshots to @TwistedAsylumMC on Discord 😅